Sources/PrimerSDK/Classes/User Interface/OAuth/OAuthViewController.swift

```diff
@@ -87,52 +86,59 @@ internal class OAuthViewController: PrimerViewController {
             (rootViewController?.presentationController as? PresentationController)?.blurEffectView.alpha = 0.7
         }
         
-        let webViewController = WebViewController()
-        webViewController.url = URL(string: urlString)
-        webViewController.delegate = self
-        webViewController.klarnaWebViewCompletion = { [weak self] (_, err) in
-            if let err = err {
-                _ = ErrorHandler.shared.handle(error: err)
-                router.show(.error(error: err))
-                
-            } else {
-                guard let host = self?.host else {
-                    let error = PrimerError.failedToLoadSession
-                    router.show(.error(error: error))
-                    return
+        if (host == .apaya) {
+            let webViewController = ApayaWebViewController()
```

Comment:
I'm not sure why we'd need another class for Apaya's webview controller. `klarnaWebviewCompletion` can be optional, and also renamed to `webviewCompletion`.

Other than that, their functionalities are the same I believe. You can just init the controller with the view model needed, then the controller will grab the `url` (or any other information needed) from the view model. 

The `completionHandler` can also be part of the view model, and be removed from the webview controller.

Reply:
Yeah! I think once we compare them we'll be able to figure out what's the best way. I want to avoid situations were these view controllers become too bloated, we should raise this in a separate ticket probably.